presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotQueryGeneratorContext.java

```diff
@@ -151,7 +154,19 @@ public PinotQueryGeneratorContext withAggregation(
             Set<VariableReferenceExpression> hiddenColumnSet)
     {
         // there is only one aggregation supported.
-        checkSupported(!hasAggregation(), "Pinot doesn't support aggregation on top of the aggregated data");
+        // checkSupported(!hasAggregation(), "Pinot doesn't support aggregation on top of the aggregated data");
+        // there is only one aggregation supported unless distinct is used.
+        Iterator<Selection> selectionIterator = newSelections.values().iterator();
+        boolean passed = false;
+        while (selectionIterator.hasNext()) {
+            if (selectionIterator.next().getDefinition().startsWith(DISTINCT_COUNT_FUNCTION_NAME.toUpperCase(Locale.ENGLISH))) {
+                hiddenColumnSet = Collections.emptySet();
```

Comment:
I didn't understand fully the need to change hiddenColumnSet here ?

Also, can this loop be written using some java8 streams way ?

Reply:
Current code will translate the underlying non-agg groupby query `select A from myTable group by A` to   `select A, count(*) from myTable group by A` and `count(*)` is hidden column.

So the upper level distinctCount shouldn't inherited this hiddenColumnSet.

Reply:
I didn't fully understand this. What distinct query is leading to a hiddenColumnSet being populated?

Perhaps you mean that a "select count(distinct a) from T", is going to show up here are as "select a from T group by T" with an additional distinct added on top? 

Is it possible for a query like: "select b, count(distinct a) from T group by b" to end up here ? ie, would we support grouped count-distinct in this PR? If not, then its okay to clear the hiddenColumnSet. I would suggest adding this comment in the code.

In any case, this whole hiddenColumnSet stuff is a sad hack and I am sorry you are having to deal with it :-(.

Reply:
For `select count(distinct origin), dest from airlineStats group by dest;`
Below is the query plan:
![image](https://user-images.githubusercontent.com/1202120/78747379-2251fe00-791e-11ea-87f8-8fc39b2972ca.png)

Expended:
![image](https://user-images.githubusercontent.com/1202120/78747500-67763000-791e-11ea-9c9e-27e4a1eec981.png)

From top to bottom, 
`AggregationNode@14771` is a non-agg `groupby dest, origin`
`AggregationNode@14754` is a `count(origin) group by dest`.

In `PinotQueryGenerator`, it will first visit `AggregationNode@14771` and generate a hidden column `count(*)`, as it's a non-agg groupby.

Then comes to `AggregationNode@14754`, we check if this is a `distinctCount` function, then we will discard the hidden column and only keep `group by dest`

Reply:
Got it. So is there a way to only discard the hiddenColumnSet iff this is a case of an aggregation atop an aggregation (as will be generated in this example query)

Also, just to confirm: we do support grouped count distinct in addition to global (non grouped) count distinct. right ?

Reply:
so this plan is common for count distinct and count distinct group by query, and we only discard hidden column under this circumstance.

Reply:
Thank you. So its safe to discard then ! :-). This is certainly worth documenting more clearly in the code.